Core/App/AppInstaller.php

```diff
@@ -356,23 +354,28 @@ private function testMysql($dbData)
     private function testPostgreSql($dbData)
     {
         $connectionStr = 'host=' . $dbData['host'] . ' port=' . $dbData['port'];
-        $connection = \pg_connect($connectionStr . ' user=' . $dbData['user'] . ' password=' . $dbData['pass']);
-        if ($connection) {
-            // Check that the DB exists, if it doesn't, we create a new one
-            $connection2 = \pg_connect($connectionStr . ' dbname=' . $dbData['name'] . ' user=' . $dbData['user'] . ' password=' . $dbData['pass']);
-            if ($connection2) {
+        $connection = @\pg_connect($connectionStr . ' dbname=postgres user=' . $dbData['user'] . ' password=' . $dbData['pass']);
+        if (is_resource($connection)) {
+            // Check that the DB exists, if it doesn't, we try to create a new one
+            $sqlExistsBD = 'SELECT 1 AS result FROM pg_database WHERE datname="' . $dbData['name'] . '"';
+            $result = \pg_query($connection, $sqlExistsBD);
+            if (is_resource($result) && \pg_num_rows($result) > 0) {
                 return true;
             }
 
-            $sqlCrearBD = 'CREATE DATABASE "' . $dbData['name'] . '";';
-            if (\pg_query($connection, $sqlCrearBD)) {
+            $sqlCreateBD = 'CREATE DATABASE "' . $dbData['name'] . '";';
+            $result = \pg_query($connection, $sqlCreateBD);
+            if (is_resource($result)) {
                 return true;
             }
-
-            $this->miniLog->critical((string) \pg_last_error($connection));
```

Comment:
In original source, this have no sense to me, if $connection exists, inside this if we are trying with $connection2, why show the last error from a connection that haven't got an error?

Reply:
Also, if first $connection fails, $connections results false, and pg_last_errors expects to receive a resource.

We can't get pg_last_error from a non resource.